tests/tests_validation_workflow/test_api_test_cases.py

```diff
@@ -0,0 +1,32 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import unittest
+from unittest.mock import Mock, call, patch
+
+from validation_workflow.api_request import ApiTest
+from validation_workflow.api_test_cases import TestCases
+
+
+class TestTestCases(unittest.TestCase):
+
+    @patch('validation_workflow.api_test_cases.ValidationArgs')
+    @patch.object(ApiTest, 'api_get')
+    def test_test_cases(self, mock_api_get: Mock, mock_validation_args: Mock) -> None:
+        mock_validation_args.return_value.stg_tag.return_value = '1.0.0.1000'
+        mock_api_get.return_value = (200, 'green')
+        test_cases = TestCases()
+        result = test_cases.test_cases()
+
+        self.assertEqual(result[1], 'There are 2/4 test cases Pass')  # type: ignore
+        mock_api_get.assert_has_calls([call(), call(), call(), call()])
```

Comment:
Not sure what are we asserting here. Can you please explain? Maybe I am missing out something.

Reply:
Thanks, @gaiksaya .. currently we defined 4 test cases in api_test_cases.py, and this assertion does two things : 1) making sure there will be 4 calls, one for each test case; 2), with test data retunrned, we expected 2 fail and 2 pass, so, we assert the exact string 'There are 2/4 test cases Pass' shown up.

Reply:
In the case you can assert the number of calls :) will make its easier for line number 26

Reply:
LOL // indeed, indeed ... why I did not think of that .. LOL